onnxruntime/core/providers/xnnpack/detail/utils.cc

```diff
@@ -17,6 +18,39 @@
 namespace onnxruntime {
 namespace xnnpack {
 
+InternalDataInitializer::InternalDataInitializer(const ONNX_NAMESPACE::TensorProto& tensor_proto) {
+  status_ = Status::OK();
+  if (!utils::HasDataType(tensor_proto)) {
+    status_ = ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Initializer must have a datatype");
+    return;
+  }
+  if (utils::HasExternalData(tensor_proto)) {
+    status_ = ORT_MAKE_STATUS(
+        ONNXRUNTIME, FAIL,
+        "don't support externalData in this helper class, please use class `Initializer` ");
+    has_external_data_ = false;
+    return;
+  }
+
+  auto proto_data_type = tensor_proto.data_type();
+  auto proto_dims = utils::GetTensorShapeFromTensorProto(tensor_proto);
+  shape_ = TensorShape(proto_dims);
+
+  type_ = DataTypeImpl::TensorTypeFromONNXEnum(proto_data_type)->GetElementType();
+
+  Path external_path;
+  status_ = utils::UnpackInitializerData(tensor_proto, external_path, unpacked_tensor_);
+  if (!status_.IsOK()) {
+    return;
+  }
+
+  const char* raw_data = static_cast<const char*>(tensor_proto.raw_data().data());
+  if (nullptr != raw_data && utils::IsPrimitiveDataType<std::string>(type_)) {
+    status_ = Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "string tensor can not have raw data");
+    return;
+  }
```

Comment:
Not quite understanding this. I would expect UnpackInitializerData to handle all these sorts of checks.

Reply:
I should put this check ahead of  `utils::UnpackInitializerData`, but UnpackInitializerData will return false in that case. removed.